flexget/plugins/input/sonarr_emit.py

```diff
@@ -0,0 +1,125 @@
+        },
+        'required': ['api_key', 'base_url'],
+        'additionalProperties': False
+    }
+
+    def get_page(self, task, config, page_number):  # Function that gets a page number and page size and returns the responding result json
+        parsedurl = urlparse(config.get('base_url'))
+        url = '%s://%s:%s%s/api/wanted/missing?page=%d&pageSize=%d&sortKey=series.title&sortdir=asc' % (parsedurl.scheme, parsedurl.netloc, config.get('port'),
+                                                                                                        parsedurl.path, page_number, config.get('page_size'))
+        headers = {'X-Api-Key': config['api_key']}
+        json = task.requests.get(url, headers=headers).json()
+        return json
+
+    def on_task_input(self, task, config):
+        json = self.get_page(task, config, 1)
+        entries = []
+        pages = int(json['totalRecords'] / config.get('page_size')) + (json['totalRecords'] % config.get('page_size') > 0)  # Sets number of requested pages
+        current_series_id = 0  # Initializes current series parameter
+        for page in range(2, pages):
+            for record in json['records']:
+                if current_series_id != record['seriesId']:
+                    current_series_id = record['seriesId']
+                    season = record['seasonNumber']
+                    episode = record['episodeNumber']
+                    entry = Entry(url='',
+                                  series_name=record['series']['title'],
+                                  series_season=season,
+                                  series_episode=episode,
+                                  series_id='S%02dE%02d' % (season, episode),
+                                  tvdb_id=record['series']['tvdbId'],
+                                  tvrage_id=record['series']['tvRageId'],
+                                  title=record['series']['title'] + ' ' + 'S%02dE%02d' % (season, episode))
+                    if entry.isvalid():
+                        entries.append(entry)
+                    else:
+                        log.debug('Invalid entry created? %s' % entry)
+                    # Test mode logging
+                    if task.options.test:
+                        log.info("Test mode. Entry includes:")
+                        log.info("    Title: %s" % entry["title"])
+                        log.info("    URL: %s" % entry["url"])
+                        log.info("    Show name: %s" % entry["series_name"])
+                        log.info("    TVDB ID: %s" % entry["tvdb_id"])
+                        log.info("    TVRAGE ID: %s" % entry["tvrage_id"])
+                        log.info("    Season: %s" % entry["series_season"])
+                        log.info("    Episode: %s" % entry["series_episode"])
+                        log.info("    Series ID: %s" % entry["series_id"])
+                    continue
+            json = self.get_page(task, config, page)
```

Comment:
You mean run get all data into local json and then operate on it in its entirety? I though of that but data can be potentially huge. Image dozens of shows, each one with several seasons of unwatched episodes.
Also note that there is a preliminary API call before the loop to get the total pages value. If there is a outstanding connection error that is derived from bad config, the plugin should stop there.